### PR TITLE
Test coverage for trunk check payload

### PR DESCRIPTION
**Pull request by jennygzma**
add test coverage for check-mode=payload.

- crafted separate payloads for each check mode: all, trunk_merge, pull_request, but have the trunkPath be a stub
- checkout out trunk-action to local-action
- craft env var TEST_GITHUB_EVENT_PATH where we store the new payload, to replace GITHUB_EVENT_PATH. we also now know based on if GITHUB_EVENT_PATH is set if we're in a test
- run local-action
- assert that the calls to CLI are correct

### CODE REVIEW

File: .github/workflows/action_tests.yaml [hunk 1]
```diff
@@ -145,3 +145,46 @@ jobs:
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: local-action
+
+      - name: Craft TEST_GITHUB_EVENT_PATH
+        shell: bash
+        run: |
+          echo "TEST_GITHUB_EVENT_PATH=local-action/${{ matrix.payload_path }}" >>$GITHUB_ENV
```
Comment (sxlijin) [inline .github/workflows/action_tests.yaml:172]: let's try to craft `TEST_GITHUB_EVENT_PATH` using `payload_path` now using htis trick:

```
jq --null-input --argfile foo foo.json '{"inputs": ($foo | tostring)}'
```

File: .github/workflows/action_tests.yaml [hunk 2]
```diff
@@ -145,3 +145,46 @@ jobs:
...
+          - description: trunk-merge-payload
+            payload_path: action_tests/trunk_merge_test_payload.json
+          - description: all-payload
+            payload_path: action_tests/all_test_payload.json
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: local-action
```
Comment (sxlijin) [inline .github/workflows/action_tests.yaml:167]: we should be able to use the src-repo / repo-under-test pattern in the other tests to bootstrap the test repo for these

**Review by sxlijin — COMMENTED** [review]
Really nice, glad to see this come together. The last big thing that I'd like to see happen here is to set up the git refs so that we exercise the git ref logic in all.sh/pull_request.sh/trunk_merge.sh - we don't want to just skip that logic if TEST_GITHUB_EVENT_PATH is set.

File: .github/workflows/action_tests.yaml (hunk 2 above)
Comment (sxlijin) [inline .github/workflows/action_tests.yaml:167]: option 1

```suggestion

      - name: create src-repo
        run: |
          cd ../
          mkdir -p src-repo
          cd src-repo
          git init
          git commit --allow-empty -m "make source repo"
      
      - name: clone into current
        run: |
          git clone ../src-repo .git clone "${src_repo}" "${output_repo}"
          
      - name: Checkout
        uses: actions/checkout@v3
        with:
          path: local-action
```

Comment (sxlijin) [reply .github/workflows/action_tests.yaml:167]:
```suggestion
      - uses: actions/checkout@v3
        with:
          path: ../local-action
        
      - run: ../local-action/setup.sh ../src-repo .        

      - name: Checkout
        uses: actions/checkout@v3
        with:
          path: local-action

      - uses: ./local-action/
```

**Comment by jennygzma** [comment]
too messy agaijn, closing

**Pull request closed without merging**